Analysis/bin/ShuffleMergeSpectral.cxx

```diff
@@ -525,27 +529,52 @@ class DataSetProcessor {
         for(TauType type: spectrum.second->ttypes)
           accumulated_entries[type] += entries_.at(type);
       }
+
+      // Added control over the ratio of tau types
+      // Step 1: for non-"-1"(take all) types:
+      // <expected ration>/<entries_per_type> is calc.
+
+      // First, check that the ratio type exists within the entries
+      for(auto tauR_: tau_ratio){
+        if (accumulated_entries.find(tauR_.first) == accumulated_entries.end()){
+          throw exception("Tau type %1% was specified in tau ratios, but no %1% is present in the input")
+            %tauR_.first; 
+        }
+      }
+
       for(auto tauR_: tau_ratio){
         if(tauR_.second!=-1){
           if(accumulated_entries.at(tauR_.first)==0)
             throw exception("No taus of the type '%1%' are found in the tuples") % tauR_.first;
+          else if(tauR_.second<0)
+            throw exception("Available --tau_ratio arguments should be > 0 or -1");
+
```

Comment:
Shouldn't we add one more fix here that you proposed:
if ( accumulated_entries.find(tauR_.first) == accumulated_entries.end())

Reply:
it's done here https://github.com/cms-tau-pog/TauMLTools/pull/41/files#diff-d1b44f30826c34ea73d4cd6c0df5d77d0387b7b7f0af33361734fbd61706bdd2R537-R544 for both tauR_.secoond ==-1 and != -1